### PR TITLE
feat: support js compilation.addInclude API

### PR DESCRIPTION
**Pull request by GiveMe-A-Name**
### Add compilation.addInclude API

- Require by Modern RSC Plugin 
https://github.com/web-infra-dev/rspack/issues/8469#issuecomment-2487040321

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: crates/rspack_binding_values/src/compilation/mod.rs [hunk 1]
```diff
@@ -577,6 +579,25 @@ impl JsCompilation {
     Ok(())
   }
 
+  #[napi(ts_args_type = "dependency: JsDependency, options: JsEntryOptions")]
+  pub fn add_include(
+    &mut self,
+    dependency: JsDependencyWrapper,
```
Comment (SyMind) [inline crates/rspack_binding_values/src/compilation/mod.rs:585]: Use JsDependency instead.

File: crates/rspack_binding_values/src/dependency.rs [hunk 2]
```diff
@@ -178,3 +178,20 @@ impl ToNapiValue for JsDependencyWrapper {
     })
   }
 }
+
+impl FromNapiValue for JsDependencyWrapper {
```
Comment (SyMind) [inline crates/rspack_binding_values/src/dependency.rs:182]: JsDependencyWrapper need not impl FromNapiValue

File: crates/rspack_binding_values/src/compilation/mod.rs (hunk 1 above)
Comment (GiveMe-A-Name) [reply crates/rspack_binding_values/src/compilation/mod.rs:585]: Got it.

Comment (GiveMe-A-Name) [reply crates/rspack_binding_values/src/compilation/mod.rs]: Done~

File: crates/rspack_binding_values/src/dependency.rs (hunk 2 above)
Comment (GiveMe-A-Name) [reply crates/rspack_binding_values/src/dependency.rs]: Done~

**Comment by SyMind** [comment]
I'd like to know what this JavaScript API is used for in RSC?

**Comment by GiveMe-A-Name** [comment]
> I'd like to know what this JavaScript API is used for in RSC?

You can see this plugin that implement in webpack 
https://github.com/web-infra-dev/modern.js/blob/feat-next-wym/packages/cli/uni-builder/src/shared/rsc/plugins/rsc-server-plugin.ts#L101-L128

**Comment by SyMind** [comment]
Firstly, I haven't found the way in which `addInclude` is used in `Next.js` and `Modern.js`.

Secondly, I think that `addInclude` is supposed to receive a `Dependency` that is manually constructed by developers on the JavaScript side. However, there isn't a relevant JavaScript API available for this at present.

**Pull request closed without merging**